inc/common/purge.php

```diff
@@ -441,7 +442,32 @@ function do_admin_post_rocket_purge_cloudflare() {
 
 	set_transient( $GLOBALS['current_user']->ID . '_cloudflare_purge_result', $cf_purge_result );
 
-	wp_redirect( wp_get_referer() );
+	wp_safe_redirect( esc_url_raw( wp_get_referer() ) );
 	die();
 }
 add_action( 'admin_post_rocket_purge_cloudflare', 'do_admin_post_rocket_purge_cloudflare' );
+
+/**
+ * Clean the cache when the current theme is updated
+ *
+ * @param WP_Upgrader $wp_upgrader WP_Upgrader instance.
+ * @param array       $hook_extra  Array of bulk item update data.
+ * @return void
+ */
+function rocket_clean_cache_theme_update( $wp_upgrader, $hook_extra ) {
+	if ( 'update' !== $hook_extra['action'] ) {
+		return;
+	}
+
+	if ( 'theme' !== $hook_extra['type'] ) {
+		return;
+	}
+
+	$current_theme = wp_get_theme();
+
+	if ( ! in_array( $current_theme->get_template(), $hook_extra['themes'], true ) ) {
```

Comment:
In case of a child theme, we should check for both parent and child.
To test, but this should do the trick:
```php
$themes = [
	$current_theme->get_template(), // Parent theme.
	$current_theme->get_stylesheet(), // Child theme.
];

if ( ! array_intersect( $hook_extra['themes'], $themes ) ) {
	return;
}
```